clients/src/main/resources/common/message/ApiVersionsResponse.json

```diff
@@ -18,14 +18,24 @@
   "type": "response",
   "name": "ApiVersionsResponse",
   // Version 1 adds throttle time to the response.
+  //
   // Starting in version 2, on quota violation, brokers send out responses before throttling.
-  "validVersions": "0-2",
+  //
+  // Version 3 is the first flexible version. Tagged fields are only supported in the body but
+  // not in the header. The length of the header must not change in order to guarantee the
+  // backward compatibility.
+  //
+  // Starting from Apache Kafka 2.4, ApiKeys field is populated with the supported versions of
+  // the ApiVersionsRequest when an UNSUPPORTED_VERSION error is returned.
+  "validVersions": "0-3",
+  // TODO(dajac) Enable this once KIP-482 is committed.
+  // "flexibleVersions": "3+",
   "fields": [
     { "name": "ErrorCode", "type": "int16", "versions": "0+",
       "about": "The top-level error code." },
     { "name": "ApiKeys", "type": "[]ApiVersionsResponseKey", "versions": "0+",
       "about": "The APIs supported by the broker.", "fields": [
-      { "name": "Index", "type": "int16", "versions": "0+", "mapKey": true,
+      { "name": "ApiKey", "type": "int16", "versions": "0+", "mapKey": true,
```

Comment:
@cmccabe Could we rename this field? `apiKey` is used everywhere at the moment and `index` feels weird. Is it consider as a public API?

Reply:
I think anything in the protocols is considered public and would need a KIP.

Reply:
I have noticed that few names have been changed in this PR #7353. The wire format, as described in the documentation, is for sure public. I am not sure about the generator part...

Reply:
@mumrah: the protocol JSON is not considered a public API.  At some point, that might change, but for now, it's still evolving and can be changed without a KIP.  The protocol itself is public, though.